dockersetup.py

```diff
@@ -0,0 +1,904 @@
+    return ret
+
+def edit_nginx_ssl_conf(hostname, https_port, certdir, certfile, keyfile):
+    filedata = readfile('docker/nginx-ssl.conf')
+    newlines = []
+    lines = filedata.splitlines()
+    for line in lines:
+        if 'ssl_certificate ' in line:
+            format = line[0:line.find('ssl_certificate')]
+            newlines.append(format + 'ssl_certificate ' + os.path.join(certdir, certfile) + ';\n')
+        elif 'ssl_certificate_key ' in line:
+            format = line[0:line.find('ssl_certificate_key')]
+            newlines.append(format + 'ssl_certificate_key ' + os.path.join(certdir, keyfile) + ';\n')
+            # Add a line for the dhparam file
+            newlines.append(format + 'ssl_dhparam ' + os.path.join(certdir, 'dhparam.pem') + ';\n')
+        elif 'https://layers.openembedded.org' in line:
+            line = line.replace('https://layers.openembedded.org', 'https://%s:%s' % (hostname, https_port))
+            newlines.append(line + "\n")
+        else:
+            line = line.replace('layers.openembedded.org', hostname)
+            newlines.append(line + "\n")
+
+    # Write to a different file so we can still replace the hostname next time
+    writefile("docker/nginx-ssl-edited.conf", ''.join(newlines))
+
+
+def edit_settings_py(emailaddr):
+    filedata = readfile('docker/settings.py')
+    newlines = []
+    lines = filedata.splitlines()
+    in_admins = False
+    for line in lines:
+        if in_admins:
+            if line.strip() == ')':
+                in_admins = False
+            continue
+        elif line.lstrip().startswith('ADMINS = ('):
+            if line.count('(') > line.count(')'):
+                in_admins = True
+            newlines.append("ADMINS = (\n")
+            if emailaddr:
+                newlines.append("  ('Admin', '%s'),\n" % emailaddr)
+            newlines.append(")\n")
+            continue
+        newlines.append(line + "\n")
+    writefile("docker/settings.py", ''.join(newlines))
+
+
+def read_dockerfile_web():
+    no_https = True
+    with open('Dockerfile.web', 'r') as f:
+        for line in f:
+            if line.startswith('COPY ') and line.rstrip().endswith('/etc/nginx/nginx.conf'):
+                if 'nginx-ssl' in line:
+                    no_https = False
+                break
+    return no_https
+
+
+def edit_dockerfile_web(hostname, no_https):
+    filedata = readfile('Dockerfile.web')
+    newlines = []
+    lines = filedata.splitlines()
+    for line in lines:
+        if line.startswith('COPY ') and line.endswith('/etc/nginx/nginx.conf'):
+            if no_https:
+                srcfile = 'docker/nginx.conf'
+            else:
+                srcfile = 'docker/nginx-ssl-edited.conf'
+            line = 'COPY %s /etc/nginx/nginx.conf' % srcfile
+        newlines.append(line + "\n")
+    writefile("Dockerfile.web", ''.join(newlines))
+
+
+def setup_https(hostname, http_port, https_port, letsencrypt, cert, cert_key, emailaddr):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this function is 19. By comparison, 98% of the functions in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the function has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code blocks on lines 476-511, lines 513-570 into separate functions.